replica/database.hh

```diff
@@ -770,10 +774,15 @@ public:
     // Applies given mutation to this column family
     // The mutation is always upgraded to current schema.
     void apply(const frozen_mutation& m, const schema_ptr& m_schema, db::rp_handle&& h = {}) {
-        do_apply(std::move(h), m, m_schema);
+        if (auto cg = single_compaction_group_if_available()) {
+            do_apply(*cg, std::move(h), m, m_schema);
+            return;
+        }
+        dht::token t = dht::get_token(*m_schema, m.key());
+        do_apply(compaction_group_for_token(t), std::move(h), m, m_schema);
```

Comment:
We can have a variant of `compaction_group_for_token` taking references to the schema and key that will hide the complexity of `single_compaction_group_if_available`, doing essentially what this function is doing, just reusable in further use cases that may need it.

Reply:
Makes sense. Can be named compaction_group_for_key().